helpers/test-helper.js

```diff
@@ -991,7 +992,12 @@ exports.getColonyEditable = async function getColonyEditable(colony, colonyNetwo
 exports.getWaitForNSubmissionsPromise = async function getWaitForNSubmissionsPromise(repCycleEthers, rootHash, nLeaves, jrh, n) {
   return new Promise(function (resolve, reject) {
     repCycleEthers.on("ReputationRootHashSubmitted", async (_miner, _hash, _nLeaves, _jrh, _entryIndex, event) => {
-      const nSubmissions = await repCycleEthers.getNSubmissionsForHash(rootHash, nLeaves, jrh);
+      let nSubmissions;
+      if (rootHash) {
```

Comment:
> Would a user pass rootHash === undefined under some circumstances?

A user would, and indeed I do in this PR! This is the difference between 'wait for n Submissions for a particular root hash submission' and 'wait for n Submissions for any root hash submission'.

> If rootHash is defined, why check the event?
I do not understand this question... can you be more specific?

Reply:
It seems strange to me that we watch for an event and then use none of the arguments (e.g. none of `_hash, _nLeaves, _jrh, _entryIndex`) because we've passed in alternatives. Why watch for the event?

Reply:
So that we know when it's worth checking `getNSubmissionsForHash`. Would you expect this to be done with `setInterval`, or something, in that scenario? We could check if `rootHash==_hash`,  and only call `getNSubmissionsForHash` if so?